src/activatable/activatable.ts

```diff
@@ -85,8 +85,22 @@ export class ActivatableCollection<T extends IMinimalActivatable> extends Activa
   public readonly activatables: T[] = [];
 
   protected async doActivate() {
-    for (const activatable of this.activatables) {
-      await activatable.activate();
+    const activated: T[] = [];
+    try {
+      for (const activatable of this.activatables) {
+        await activatable.activate();
+        activated.push(activatable);
```

Comment:
could just unshift, so no reverse is needed.

Reply:
I thought about that. But I suspect that `push` is faster, and since most of the time we don't need to reverse and deactivate, I figure this is actually ever so slightly faster on average.

Reply:
and still easy to understand